### ISSUE TITLE
can you please update for huggingface?  (specifically interested to getting it working on rtx 3090)

### ISSUE THREAD

**Issue by johndpope** [paulcjh/gpt-j-6b#9]
I have this drafted code - 

```

from transformers import GPTJForCausalLM, AutoTokenizer
import torch
import os
import time


## CUSTOM USER INPUT
prompt = "Create a new song title a new band name. Be creative! " \
"###" \
"Band name: The Execs" \
"Song title: Company Meeting" \
"###" \
"Band name: The One Chords" \
"Song title: Moving Down to Burlington" \
"###" \
"Band name."
temperature = 0.9
max_length = 100
## END CUSTOM USER INPUT


# Load cached model+tokenizer
load_start_time = time.time()
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
modelpath = os.path.join(os.getcwd(), "saved_files\\gptj",  'gptj-model.pth')

# I created a subdirectory / and cloned 24gb LFS repo to gtp-j-6B
# am running on RTX - 3090 (but only 32gb of cpu RAM)
tokenizer = AutoTokenizer.from_pretrained("./EleutherAI/gpt-j-6B")
model = GPTJForCausalLM.from_pretrained("./EleutherAI/gpt-j-6B", revision="float16", low_cpu_mem_usage=True)
print("Load completed in - ", time.time() - load_start_time)
# If model is not yet saved - execute this code instead of above code
#model = GPTJForCausalLM.from_pretrained("EleutherAI/gpt-j-6B",  torch_dtype=torch.float16)
model = model.to(device)
#torch.save(model, modelpath)

generate_start_time = time.time()
input_ids = tokenizer(prompt, return_tensors="pt").input_ids.to(device)
gen_tokens = model.generate(input_ids, do_sample=True, temperature=temperature, max_length=max_length)
gen_text = tokenizer.batch_decode(gen_tokens)[0]
print("Generation completed in - ", time.time() - generate_start_time)
print("--------------------------------")
print(gen_text)
print("--------------------------------")

```

It seems that one needs 64GB / CPU memory....
https://huggingface.co/transformers/master/model_doc/gptj.html
_The model should fit on 16GB GPU for inference. For training/fine-tuning it would take much more GPU RAM. Adam optimizer for example makes four copies of the model: model, gradients, average and squared average of the gradients. So it would need at least 4x model size GPU memory, even with mixed precision as gradient updates are in fp32._

**Comment by johndpope** [comment]
found this - https://gist.githubusercontent.com/kinoc/f3225092092e07b843e3a2798f7b3986/raw/fc0dbe522d09d3797dd2a64e7182003f7d9a7fa8/jserv.py
